src/main/java/com/mparticle/kits/ButtonKit.java

```diff
@@ -170,6 +178,51 @@ public void onResult(@Nullable Intent intent, @Nullable Throwable throwable) {
         return null;
     }
 
+    /*
+     * Overrides for CommerceListener
+     */
+
+    @Override
+    public List<ReportingMessage> logLtvIncrease(BigDecimal bigDecimal, BigDecimal bigDecimal1,
+            String s, Map<String, String> map) {
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> logEvent(CommerceEvent commerceEvent) {
+        if (commerceEvent == null
+                || commerceEvent.getProductAction() == null
+                || commerceEvent.getProducts() == null) {
+            return null;
+        }
+
+        List<ButtonProductCompatible> products = parseAsButtonProducts(commerceEvent.getProducts());
+        switch (commerceEvent.getProductAction()) {
+            case Product.DETAIL:
+                // Since the ML does not support batch queuing, we will report a max of 10 products
+                for (int i = 0; i < 10 && i < products.size(); i++) {
+                    logDebug("Tracking product viewed: %s", products.get(i).getName());
+                    merchant.trackProductViewed(products.get(i));
+                }
+                break;
+            case Product.ADD_TO_CART:
+                // Since the ML does not support batch queuing, we will report a max of 10 products
+                for (int i = 0; i < 10 && i < products.size(); i++) {
+                    logDebug("Tracking product added to cart: %s", products.get(i).getName());
+                    merchant.trackAddToCart(products.get(i));
+                }
```

Comment:
@wessmith Thoughts on introducing a maximum? I'd wager that most times the list size will be 1, but considering I get back a list I figured it's best to include a reasonable amount.

Reply:
Discussed offline: limiting to 2 and including `btn_product_count` attribute to every product with the value being the size of the product list